tests/integration/test_object_storage.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from types import SimpleNamespace
+
+import pytest
+import yaml
+from helpers import get_address, oci_image
+from pytest_operator.plugin import OpsTest
+from workload import Mimir
+
+logger = logging.getLogger(__name__)
+
+mimir = SimpleNamespace(
+    name="mimir", resources={"mimir-image": oci_image("./metadata.yaml", "mimir-image")}
+)
+tester = SimpleNamespace(charm="avalanche-k8s", name="avalanche")
+storage = SimpleNamespace(charm="s3proxy-k8s", name="s3proxy")
+
+
+@pytest.mark.abort_on_fail
+async def test_deploy_and_relate_charms(ops_test: OpsTest, mimir_charm):
+    """Build up the model for verification."""
+    await asyncio.gather(
+        ops_test.model.deploy(
+            mimir_charm,
+            resources=mimir.resources,
+            application_name=mimir.name,
+            trust=True,
+        ),
+        ops_test.model.deploy(
+            tester.charm,
+            application_name=tester.name,
+            channel="edge",
+        ),
+        ops_test.model.deploy(storage.charm, application_name=storage.name, channel="edge"),
```

Comment:
How come you don't need to config it?
Can we have an indirect way to confirm our config in results in mimir attempting to user object storage?

Reply:
It's preconfigured. If no key/secret is set, it generates one. Is there a way other than the actual running Mimir config (via endpoint) that you feel would be a better way of confirming it?

Reply:
Sure, if you can confirm from an endpoint that mimir is doing s3 now then great.